backend/src/main/java/com/empire/EntryServlet.java

```diff
@@ -259,6 +259,14 @@ private String getAdvancePoll() {
 						for (String mail : emails.keySet()) {
 							mail(mail, "👑 Empire: Turn Advances", emails.get(mail).replace("%GAMEID%", "" + gameId));
 						}
+						if (w.gameover) {
```

Comment:
Could this whole creation of newActiveGames not be made a little simpler with the static creation method already existing in the ActiveGames class?

Reply:
Yes, this was needlessly complicated; I've been working with immutable objects so much in other stuff I forgot I didn't need to make a copy here. Cleaned up a bit.

The lack of constructors is an artifact of using Gson so heavily in all these classes. Gson uses reflection and various trickery to construct the object and initialize its fields (even private fields) when you call fromJson(), and uses the no-arguments constructor if it exists (and skips using a constructor at all if it doesn't). The no-arguments constructor exists by default, unless you define another one, so I've been avoiding defining constructors in classes we build from JSON mostly to remind myself that we can't depend on any particular constructor's contracts.

In many cases, there's also good reasons to prefer using static factory methods instead of constructors, but typically in that case you would see private constructors defined to prevent other classes from calling them. I think we actually could do that here (I think Gson doesn't care if the no-args constructor is private?), but I haven't really settled this issue with Gson thrown into the mix completely in my mind.